homeassistant/components/amcrest/camera.py

```diff
@@ -387,7 +387,9 @@ def update(self) -> None:
                 else:
                     self._model = "unknown"
             if self._attr_unique_id is None:
-                self._attr_unique_id = self._api.serial_number.strip()
+                self._attr_unique_id = (
```

Comment:
How will this work with existing configurations, I think you need to add some code that updates existing self._attr_unique_id.

Reply:
How would that be done? I'm not sure what changes are needed to update an existing unique id.

Reply:
I don't think it can make it upgrade hass automatically. But you could mark the change as breaking.

Reply:
It is possible, delete the old config entry and automatically make a new one.

You cannot update the unique id, but you can save people from having to manually delete and create a new configuration entry.

Reply:
That is great, do you know where that would need to be done or if there is an example of this somewhere else? Is that something that would need to be handled in `async_setup_platform`?